### PR TITLE
Feature/trust person

### PR DESCRIPTION
**Pull request by s20383**
testy dodam później

### CODE REVIEW

File: src/main/java/com/gary/backendv2/service/TrustedPersonService.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	public void deleteByEmail(String userEmail){
+		Optional<User> userOptional = userRepository.findByEmail(userEmail);
+		if (userOptional.isEmpty()) {
+			throw new HttpException(HttpStatus.NOT_FOUND, String.format("Cannot find user with %s", userEmail));
+		}
+		TrustedPerson trustedPerson = userOptional.get().getTrustedPerson();
+		if(trustedPerson == null){
+			throw new HttpException(HttpStatus.NO_CONTENT, String.format("User %s don't have trusted person", userEmail));
```
Comment (s20462) [inline src/main/java/com/gary/backendv2/service/TrustedPersonService.java:69]: _The HTTP 204 No Content success status response code indicates that a request has succeeded, but that the client doesn't need to navigate away from its current page._

404 Not Found pls

File: src/main/java/com/gary/backendv2/service/TrustedPersonService.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	public void updateTrustedPerson(TrustedPersonRequest trustedPersonRequest){
+		Optional<User> userOptional = userRepository.findByEmail(trustedPersonRequest.getUserEmail());
+		if (userOptional.isEmpty()) {
+			throw new HttpException(HttpStatus.NOT_FOUND, String.format("Cannot find user with %s", trustedPersonRequest.getUserEmail()));
+		}
+		TrustedPerson trustedPerson = userOptional.get().getTrustedPerson();
+		if(trustedPerson == null){
+			throw new HttpException(HttpStatus.NO_CONTENT, String.format("User %s don't have trusted person", trustedPersonRequest.getUserEmail()));
```
Comment (s20462) [inline src/main/java/com/gary/backendv2/service/TrustedPersonService.java:106]: 404

File: src/main/java/com/gary/backendv2/service/TrustedPersonService.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+	public TrustedPersonResponse getByEmail(String userEmail){
+		Optional<User> userOptional = userRepository.findByEmail(userEmail);
+		if (userOptional.isEmpty()) {
+			throw new HttpException(HttpStatus.NOT_FOUND, String.format("Cannot find user with %s", userEmail));
+		}
+		TrustedPerson trustedPerson = userOptional.get().getTrustedPerson();
+		if(trustedPerson == null){
+			throw new HttpException(HttpStatus.NO_CONTENT, String.format("User %s don't have trusted person", userEmail));
+		}
```
Comment (s20462) [inline src/main/java/com/gary/backendv2/service/TrustedPersonService.java:51]: The HTTP 204 No Content success status response code indicates that a request has succeeded, but that the client doesn't need to navigate away from its current page.

404 Not Found pls

File: src/main/java/com/gary/backendv2/service/TrustedPersonService.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+	public void addTrustedPerson(TrustedPersonRequest trustedPersonRequest){
+		Optional<User> userOptional = userRepository.findByEmail(trustedPersonRequest.getUserEmail());
+		if (userOptional.isEmpty()) {
+			throw new HttpException(HttpStatus.NOT_FOUND, String.format("Cannot find user with %s", trustedPersonRequest.getUserEmail()));
+		}
+		User user = userOptional.get();
+		if(user.getTrustedPerson() != null){
+			throw new HttpException(HttpStatus.BAD_REQUEST, String.format("%s already have trusted person", trustedPersonRequest.getUserEmail()));
+		}
```
Comment (s20462) [inline src/main/java/com/gary/backendv2/service/TrustedPersonService.java:85]: The HTTP 204 No Content success status response code indicates that a request has succeeded, but that the client doesn't need to navigate away from its current page.

404 Not Found pls

File: src/main/java/com/gary/backendv2/model/dto/request/TrustedPersonRequest.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	private String userEmail;
+
+	@NotBlank
+	private String firstName;
+
+	@NotBlank
+	private String lastName;
+
+	@Email
+	@Nullable
```
Comment (s20462) [inline src/main/java/com/gary/backendv2/model/dto/request/TrustedPersonRequest.java:25]: czemu nullable?

nie jestem pewnien czy null albo pusty string przejdzie walidację emaila i moze być tak, że walidator nigdy nie przyjmie rządania z brakiem tego pola

**Review by s20462 — CHANGES_REQUESTED** [review]

File: src/main/java/com/gary/backendv2/service/TrustedPersonService.java (hunk 3 above)
Comment (s20462) [reply src/main/java/com/gary/backendv2/service/TrustedPersonService.java:51]: ewentualnie 200 OK ale bez response body

File: src/main/java/com/gary/backendv2/service/TrustedPersonService.java (hunk 1 above)
Comment (s20462) [reply src/main/java/com/gary/backendv2/service/TrustedPersonService.java:69]: ewentualnie 200 OK ale bez response body

Comment (s20462) [reply src/main/java/com/gary/backendv2/service/TrustedPersonService.java:69]: 10.2.5 204 No Content
The server has fulfilled the request but does not need to return an entity-body, and might want to return updated metainformation. The response MAY include new or updated metainformation in the form of entity-headers, which if present SHOULD be associated with the requested variant.

If the client is a user agent, it SHOULD NOT change its document view from that which caused the request to be sent. This response is primarily intended to allow input for actions to take place without causing a change to the user agent's active document view, although any new or updated metainformation SHOULD be applied to the document currently in the user agent's active view.

The 204 response MUST NOT include a message-body, and thus is always terminated by the first empty line after the header fields.

jednak 204 tu pasuje ale bez body, sam status HTTP as per RFC-2616

Comment (s20462) [reply src/main/java/com/gary/backendv2/service/TrustedPersonService.java:69]: no i w sumie pytanie czy chcemy być w 100% zgodni ze standardem http

File: src/main/java/com/gary/backendv2/model/dto/request/TrustedPersonRequest.java (hunk 5 above)
Comment (s20383) [reply src/main/java/com/gary/backendv2/model/dto/request/TrustedPersonRequest.java:25]: Z tego co pamiętam to u Adama było to zrobione jako opcjonalne.
Dam Unit test do tego

File: src/main/java/com/gary/backendv2/service/TrustedPersonService.java (hunk 1 above)
Comment (s20383) [reply src/main/java/com/gary/backendv2/service/TrustedPersonService.java:69]: Nwm, przy x zapytaniach wysyłanych na raz fajnie jest mieć body które lepiej opisuje o co chodziło. Mogę zmienić na inny response taki który ma body by być zgodnym ze standardem - w ostateczności mogę dać not found

Comment (s20462) [reply src/main/java/com/gary/backendv2/service/TrustedPersonService.java:69]: zostaw jak jest w sumie

**Pull request merged**